dotnet/src/Connectors/Connectors.AssemblyAI/AudioToText/AssemblyAIAudioToTextService.cs

```diff
@@ -0,0 +1,282 @@
+        }
+
+        using var request = HttpRequest.CreatePostRequest(url, jsonRequest);
+        this.AddDefaultHeaders(request);
+        using var response = await this.SendWithSuccessCheckAsync(this._httpClient, request, ct).ConfigureAwait(false);
+        var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+        var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+        if (json.RootElement.TryGetProperty("error", out var property))
+        {
+            throw new KernelException($"Failed to create transcript. Reason: {property.GetString()!}");
+        }
+
+        return json.RootElement.GetProperty("id").GetString()!;
+    }
+
+    private async Task<JsonDocument> WaitForTranscriptToProcessAsync(
+        string transcriptId,
+        PromptExecutionSettings? executionSettings = null,
+        CancellationToken ct = default
+    )
+    {
+        var url = this.Url($"v2/transcript/{transcriptId}");
+        var pollingInterval = TimeSpan.FromSeconds(1);
+        if (executionSettings is AssemblyAIAudioToTextExecutionSettings aaiSettings)
+        {
+            pollingInterval = aaiSettings.PollingInterval;
+        }
+
+        while (!ct.IsCancellationRequested)
+        {
+            using var request = HttpRequest.CreateGetRequest(url);
+            this.AddDefaultHeaders(request);
+            using var response = await this.SendWithSuccessCheckAsync(this._httpClient, request, ct).ConfigureAwait(false);
+            var jsonStream = await response.Content.ReadAsStreamAsync().ConfigureAwait(false);
+            var json = await JsonDocument.ParseAsync(jsonStream, cancellationToken: ct).ConfigureAwait(false);
+
+            var status = json.RootElement.GetProperty("status").GetString()!;
+            switch (status)
+            {
+                case "processing":
+                case "queued":
+                    await Task.Delay(pollingInterval, ct).ConfigureAwait(false);
+                    break;
+                case "completed":
+                    return json;
+                case "error":
+                    var errorString = json.RootElement.GetProperty("error").GetString()!;
+                    throw new KernelException($"Failed to create transcript. Reason: {errorString}");
+                default:
+                    throw new KernelException("Unexpected transcript status. This code shouldn't be reachable.");
+            }
+        }
+
+        ct.ThrowIfCancellationRequested();
+        throw new KernelException("This code is unreachable.");
+    }
+
+    /// <summary>
+    /// Create a URL string that includes the default BaseUrl if the BaseAddress on _httpClient isn't set.
+    /// </summary>
+    /// <param name="url">URL without base.</param>
+    /// <returns>URL with or without BaseUrl.</returns>
+    private string Url(string url)
+    {
+        return this._httpClient.BaseAddress is null ? $"{BaseUrl}{url}" : url;
```

Comment:
I think you wanted to combine _httpClient.BaseAdress with the url and not return the url itself?

Reply:
If there's a BaseAddress configured on the HTTP client, then we don't need to add the const `BaseUrl`.
`Url()` returns `https://api.assemblyai.com/path` if `this._httpClient.BaseAddress is null`
otherwise it returns `path`.